sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2244 @@
+  {
+    Set,
+    Modify,
+    Remove,
+    Unknown
+  };
+
+  std::string PathSetAccessControlRecursiveModeToString(const PathSetAccessControlRecursiveMode& pathSetAccessControlRecursiveMode)
+  {
+    switch (pathSetAccessControlRecursiveMode)
+    {
+      case PathSetAccessControlRecursiveMode::Set:
+        return "set";
+      case PathSetAccessControlRecursiveMode::Modify:
+        return "modify";
+      case PathSetAccessControlRecursiveMode::Remove:
+        return "remove";
+      default:
+        return std::string();
+    }
+  }
+
+  PathSetAccessControlRecursiveMode PathSetAccessControlRecursiveModeFromString(const std::string& pathSetAccessControlRecursiveMode)
+  {
+    if (pathSetAccessControlRecursiveMode == "set")
+    {
+      return PathSetAccessControlRecursiveMode::Set;
+    }
+    if (pathSetAccessControlRecursiveMode == "modify")
+    {
+      return PathSetAccessControlRecursiveMode::Modify;
+    }
+    if (pathSetAccessControlRecursiveMode == "remove")
+    {
+      return PathSetAccessControlRecursiveMode::Remove;
+    }
+    throw std::runtime_error("Cannot convert " + pathSetAccessControlRecursiveMode + " to PathSetAccessControlRecursiveMode");
+  };
+
+  struct AclFailedEntry
+  {
+    std::string Name;
+    std::string Type;
+    std::string ErrorMessage;
+
+    static AclFailedEntry CreateFromJson(const nlohmann::json& node)
+    {
+      AclFailedEntry result;
+      result.Name = node["name"].get<std::string>();
+      result.Type = node["type"].get<std::string>();
+      result.ErrorMessage = node["errorMessage"].get<std::string>();
+      return result;
+    }
+  };
+
+  struct SetAccessControlRecursiveResponse
+  {
+    uint32_t DirectoriesSuccessful;
+    uint32_t FilesSuccessful;
+    uint32_t FailureCount;
+    std::vector<AclFailedEntry> FailedEntries;
+
+    static SetAccessControlRecursiveResponse CreateFromJson(const nlohmann::json& node)
+    {
+      SetAccessControlRecursiveResponse result;
+      result.DirectoriesSuccessful = static_cast<uint32_t>(std::stoul(node["directoriesSuccessful"].get<std::string>()));
+      result.FilesSuccessful = static_cast<uint32_t>(std::stoul(node["filesSuccessful"].get<std::string>()));
+      result.FailureCount = static_cast<uint32_t>(std::stoul(node["failureCount"].get<std::string>()));
+      for (const auto& element : node["failedEntries"])
+      {
+        result.FailedEntries.emplace_back(AclFailedEntry::CreateFromJson(element));
+      }
+      return result;
+    }
+  };
+
+  struct Path
+  {
+    std::string Name;
+    bool IsDirectory;
+    std::string LastModified;
+    std::string ETag;
+    uint64_t ContentLength;
+    std::string Owner;
+    std::string Group;
+    std::string Permissions;
+
+    static Path CreateFromJson(const nlohmann::json& node)
+    {
+      Path result;
+      result.Name = node["name"].get<std::string>();
+      result.IsDirectory = static_cast<bool>(stoi(node["isDirectory"].get<std::string>()));
```

Comment:
I don't think this conversion should be cast

Reply:
This is actually a bug. Changed.